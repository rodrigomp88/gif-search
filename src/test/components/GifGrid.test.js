import { shallow } from "enzyme/build";
import { GifGrid } from "../../components/GifGrid";
import { useFetchGifs } from "../../hooks/useFetchGifs";
jest.mock("../../hooks/useFetchGifs.js");

describe("Pruebas en el gifGrid", () => {
  const category = "Drago Ball";

  test("debe de mostrarse correcatmente", () => {
    useFetchGifs.mockReturnValue({
      data: [],
      loading: true,
    });
    const wrapper = shallow(<GifGrid category={category} />);
    expect(wrapper).toMatchSnapshot();
  });
  test("debe de mostrar items cunado se cargan imagenes userFetchGifs", () => {
    const gifs = [
      {
        id: "ABC",
        url: "https://localhost/cualquiercosa.jpg",
        title: "cualquier cosa",
      },
    ];
    useFetchGifs.mockReturnValue({
      data: gifs,
      loading: false,
    });
    const wrapper = shallow(<GifGrid category={category} />);

    // expect(wrapper).toMatchSnapshot();

    expect(wrapper.find("p").exists()).toBe(false);

    expect(wrapper.find('GifGridItem').length).toBe(gifs.length)
  });
});
