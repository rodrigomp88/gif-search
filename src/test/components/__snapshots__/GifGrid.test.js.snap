// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el gifGrid debe de mostrarse correcatmente 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__bounce animate__fadeIn"
  >
    Drago Ball
  </h3>
  <p
    className="animate__animated animate__bounce animate__flash"
  >
    Loading
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
